src/mscorlib/shared/System/IO/Path.Windows.cs

```diff
@@ -87,6 +87,54 @@ public static string GetFullPath(string path)
             return fullPath;
         }
 
+        public static string GetFullPath(string path, string basePath)
+        {
+            if (path == null)
+                throw new ArgumentException(nameof(path));
+
+            if (basePath == null)
+                throw new ArgumentNullException(nameof(basePath));
+
+            if (!IsPathFullyQualified(basePath))
+                throw new ArgumentException(SR.Arg_BasePathNotFullyQualified);
+
+            if (basePath.Contains('\0') || path.Contains('\0'))
+                throw new ArgumentException(SR.Argument_InvalidPathChars);
+
+            if (IsPathFullyQualified(path))
+                return GetFullPath(path);
+
+            int length = path.Length;
+
+            if (PathInternal.IsDevice(basePath))
+            {
+                // paths starting with \\?\ & \\.\
+                return RemoveRelativeSegments(CombineNoChecks(basePath, path), PathInternal.GetRootLength(basePath));
+            }
+            else if ((length >= 1 && PathInternal.IsDirectorySeparator(path[0])))
+            {
+                // path is current drive rooted i.e. starts with \
+                return RemoveRelativeSegments(CombineNoChecks(GetPathRoot(basePath), path.Substring(1)));
+            }
+            else if (length >= 2 && PathInternal.IsValidDriveChar(path[0]) && path[1] == PathInternal.VolumeSeparatorChar)
+            {
+                Debug.Assert(length == 2 || !PathInternal.IsDirectorySeparator(path[2]));
+                
+                // same and different specific drive rooted path
+                if (GetPathRoot(path.AsReadOnlySpan()) == GetPathRoot(basePath.AsReadOnlySpan()))
+                {
+                    // paths such as C:foo
+                    return RemoveRelativeSegments(CombineNoChecks(basePath, path.Substring(2)), basePath.Length);
+                }
+                else
+                {
+                    return RemoveRelativeSegments(path.Insert(2, "\\"), PathInternal.GetRootLength(path));
```

Comment:
As above, use GetFullPath on the result, not RemoveRelativeSegments. This case is `C:Foo` and `D:\Bar`, as they don't match, you're returning `C:\Foo`.

Reply:
what should be the correct output in this case ?

Reply:
what should be the correct output in this case ?

Reply:
Everything is covered by the larger code snippet I included above.

Reply:
Yes, the correct answer is C:\Foo. I didn't mean to infer that the code was returning the wrong answer, just describing the correct behavior in examples.